src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs

```diff
@@ -673,9 +673,29 @@ private void WritePositionTaggedLiteral(string value, int position)
             EndContext();
         }
 
-        public abstract void BeginContext(int position, int length, bool isLiteral);
+        public void BeginContext(int position, int length, bool isLiteral)
+        {
+            // Inlinable fast-path check if Diagnositcs is enabled
+            if (DiagnosticSource?.IsEnabled() == true)
```

Comment:
>DiagnosticSource [](start = 16, length = 16)

For instance if we used a field here that was set when `DiagnosticSource` was set, that would remove all of the breaking changes but accomplish the same thing.